pgml-dashboard/src/api/cms.rs

```diff
@@ -61,35 +62,66 @@ lazy_static! {
     );
 }
 
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum DocType {
+    Blog,
+    Docs,
+    Careers,
+}
+
+impl FromStr for DocType {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<DocType, Self::Err> {
+        match s {
+            "blog" => Ok(DocType::Blog),
+            "Doc" => Ok(DocType::Docs),
+            "Careers" => Ok(DocType::Careers),
+            _ => Err(()),
+        }
+    }
+}
+
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Document {
     /// The absolute path on disk
     pub path: PathBuf,
     pub description: Option<String>,
+    pub author: Option<String>,
+    pub author_image: Option<String>,
+    pub featured: bool,
+    pub date: Option<chrono::NaiveDate>,
+    pub tags: Vec<String>,
     pub image: Option<String>,
     pub title: String,
     pub toc_links: Vec<TocLink>,
-    pub html: String,
+    pub contents: String,
+    pub doc_type: Option<DocType>,
 }
 
+// Gets document markdown
 impl Document {
     pub async fn from_path(path: &PathBuf) -> anyhow::Result<Document, std::io::Error> {
         warn!("path: {:?}", path);
+
```

Comment:
Never noticed the ^ `warn!` above. It's a bit too noisy for production, could we move that to `debug!`?

Reply:
yeah, I can do that in my next merge.